### DISCUSSION TITLE
Unable to get results from related documents in FastAPI-Users and Beanie

### DISCUSSION THREAD

**Discussion by JimScope** [fastapi-users/fastapi-users#1006] [Q&A] [answered]
I have a couple of models:

```
class Message(Document):
    phone: str
    message_body: str
    user: Indexed(PydanticObjectId)

    class Settings:
        name = "Message"


class User(BeanieBaseUser[PydanticObjectId]):
    messages: Optional[List[Link[Message]]]
```

They have a relationship using Beanie Link. The Message model is a custom Model and User is the default of FastAPI-Users.

When I save a record with insert() on a Message that is the child model, it does not establish a relationship with the parent. Returning the dict with null messages field. Any idea what could be the error?

**Comment by frankie567** [comment, marked as answer]
I think it's because Beanie doesn't fetch linked documents by default, you have to explicitly do that: https://roman-right.github.io/beanie/tutorial/relations/#fetch

**Reply by JimScope** [reply]
I was able to review it the other day. The situation is How could I do it in a clean way with the Current User getting the related documents?

**Reply by frankie567** [reply]
Well, if you are implementing an API to retrieve the user messages, you could do this:

```py
@app.get("/messages")
async def list_messages(user: User = Depends(current_active_user)):
    await user.fetch_link(User.messages)
    return user.messages
```

However, this may not be the best option because you'll likely have a lot of messages for a single user, and probably need pagination. I'm not a super expert in MongoDB but intuitively, I would say it's better to put the relation on Message, so you can easily query messages by user.

```py
class Message(Document):
    content: str
    user: Link[User]


@app.get("/messages")
async def list_messages(limit: int = Query(10), skip: int = Query(0), user: User = Depends(current_active_user)):
    messages = await Message.find(Message.user.id == user.id).limit(limit).skip(skip).to_list()
    return messages
```

**Reply by JimScope** [reply]
Thanks a lot!
